parachain/test-parachains/undying/collator/src/lib.rs

```diff
@@ -56,19 +65,32 @@ fn calculate_head_and_state_for_number(
 		*grave = i as u8;
 	});
 
+	log::debug!(target: LOG_TARGET, "received messages from {} parachains", inbound_messages.len());
+	let inbound_hrmp_messages = inbound_messages.values().flatten().cloned().collect::<Vec<_>>();
+	log::debug!(target: LOG_TARGET, "received {} messages in total", inbound_messages.len());
+
 	let mut state = GraveyardState { index, graveyard, zombies, seal };
 	let mut head =
 		HeadData { number: 0, parent_hash: Hash::default().into(), post_state: hash_state(&state) };
+	let mut messages: Vec<OutboundHrmpMessage<ParaId>> = Vec::new();
 
 	while head.number < number {
-		let block = BlockData { state, tombstones: 1_000, iterations: pvf_complexity };
-		let (new_head, new_state) =
+		let block = BlockData {
+			state,
```

Comment:
With this changes the PoV size will no longer respect the `pov_size` configured if we send/receive messages. We should rework that part to account for this.

Reply:
That is true, but the problem is that it might be an unexpected change, as we set PoV size equal to an artificial state (graveyard). If we send/receive messages it becomes probabilistic by nature (as we don't know how many messages will be sent to us).